06.wc/wc.rb

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = {}
+  optparse = OptionParser.new { |opts| opts.on('-l', '--lines', 'count text’s lines') { options[:l] = true } }
+
+  optparse.parse!(ARGV)
+
+  ARGV == [] ? print_count(count(readlines.join, options)) : display_count_file(ARGV, options)
+end
+
+def display_count_file(argv, options)
+  total_count = Array.new(options[:l] ? 1 : 3, 0)
+  argv.each do |arg|
+    if FileTest.file?(arg)
+      current_count = total_count
+      text_count = count(File.read(arg), options)
+      print_count(text_count, arg)
+      total_count = current_count.zip(text_count).map(&:sum) if argv.count > 1
```

Comment:
[MUST]current_count に代入しないで、l23 で total_count に対して直接 zip ~ したもので total_count に入れ直す、としたほうが効率が良さそうです。

```suggestion
      text_count = count(File.read(arg), options)
      print_count(text_count, arg)
      total_count = total_count.zip(text_count).map(&:sum) if argv.count > 1
```